src/vm/lib/metadata/agent.js

```diff
@@ -749,14 +840,15 @@ MetadataAgent.prototype.createKVMServer = function (zopts, callback) {
         // When the stream closes, we'll delete from zoneConnections so that on
         // next boot (or periodic scan if for some reason we got closed while
         // the zone was actually running) we re-create.
-        zlog.info('stream closed');
+        zlog.info('stream closed on fd %d', fd);
         delete self.zoneConnections[zopts.zone];
     });
 
     kvmstream.connect(zopts.sockpath);
 
-    zlog.info('listening on fd ' + kvmstream._handle.fd);
-    self.zoneConnections[zopts.zone].fd = kvmstream._handle.fd;
+    fd = kvmstream._handle.fd;
+    zlog.info('listening on fd %d', fd);
+    self.zoneConnections[zopts.zone].fd = fd;
```

Comment:
Why do we explicitly create a `fd` property on the object referenced by `self.zoneConnections[zopts.zone]`? I would think that `self.zoneConnections[zopts.zone]` points to an object with a `conn` property that is an instance of `net.Socket`, which thus has a `._handle.fd` property. But I may be missing something.